### PR TITLE
Pretty-print does better with conveying field value structure

### LINKED ISSUE #245: eliot-prettyprint should pretty print values
Currently field values are merely stringified. Using a pretty printer (`json.dumps` or `pformat`) would lead to more readable output.

### PR DESCRIPTION
**Pull request by itamarst**
Fixes #245.

### CODE REVIEW

File: eliot/_util.py [hunk 1]
```diff
@@ -37,3 +39,22 @@ def saferepr(o):
...
+
+    @param str name: The name of the new module.
+    @param original_module: The original module we're recreating.
+
+    @return: A new, distinct module.
+    """
+    module = ModuleType(name)
+    path = original_module.__file__
+    if path.endswith(".pyc") or path.endswith(".pyo"):
+        path = path[:-1]
```
Comment (wallrj) [inline eliot/_util.py]: :question: I don't think you can assume that the `.py` file is always alongside the `.pyc` or `.pyo`. I seem to remember that Ubuntu puts them in a separate directory (in Python3 only??).
And  there may be cases where the py files are not distributed at all.
But reading down I see that this is only used for patching the traceback module....and that it's a refactoring of code that already existed and had already been reviewed. So fine.

Comment (itamarst) [inline eliot/_util.py]: On Python 3 it looks like `__file__` is always the .py, at least on Ubuntu.

Comment (itamarst) [inline eliot/_util.py]: I wonder if I can do something with `imp`, or if I rejected it for any reason.

Comment (itamarst) [inline eliot/_util.py]: Oh, interesting, yes I can use `imp`.

**Comment by wallrj** [comment]
Thanks @itamarst 

 * All the tests pass locally
 * I like the new multiline format

```
 9c360485-253b-4dc9-b9f0-c3167f89d79f -> /2/2/2/2/2/2/2/2/2/4/2/2/2/2
 2015-11-30 22:47:04.887236Z
-  action_type: fsm:transition
-  action_status: succeeded
-  _HOSTNAME: 119.9.108.114
-  _SYSTEMD_UNIT: flocker-container-agent
-  fsm_next_state: <ConvergenceLoopStates=SLEEPING>
-  fsm_output: [u'<ConvergenceLoopOutputs=STORE_INFO>', u'<ConvergenceLoopOutputs=UPDATE_MAYBE_WAKEUP>']
+  action_type: 'fsm:transition'
+  action_status: 'succeeded'
+  _HOSTNAME: '119.9.108.114'
+  _SYSTEMD_UNIT: 'flocker-container-agent'
+  fsm_next_state: '<ConvergenceLoopStates=SLEEPING>'
+  fsm_output: ['<ConvergenceLoopOutputs=STORE_INFO>',
+            |  '<ConvergenceLoopOutputs=UPDATE_MAYBE_WAKEUP>']

```

 * I noted that eliot-prettyprint breaks when there are non-eliot lines in the input.

```
4b7139d8-1ebc-4a07-a804-63fe0a536fee -> /1
2015-11-30 22:47:03.082205Z
  message_type: 'twisted:log'
  _HOSTNAME: '119.9.108.173'
  _SYSTEMD_UNIT: 'flocker-docker-plugin'
  error: False
  message: 'Stopping factory <twisted.web.client._HTTP11ClientFactory instance at 0x7f1b8da41518>'

Traceback (most recent call last):
  File "/home/richard/.virtualenvs/eliot-245/bin/eliot-prettyprint", line 9, in <module>
    load_entry_point('eliot==0.10.1+11.g61a813a', 'console_scripts', 'eliot-prettyprint')()
  File "/home/richard/projects/HybridLogic/eliot/eliot/prettyprint.py", line 99, in _main
    result = pretty_format(message) + "\n"
  File "/home/richard/projects/HybridLogic/eliot/eliot/prettyprint.py", line 67, in pretty_format
    level = "/" + "/".join(map(unicode, message[TASK_LEVEL_FIELD]))
KeyError: u'task_level'
```

It choked on a Docker log line.

It'd be nice if it printed the failed lines to stderr.
Or if it skipped over non-eliot lines.

Please merge unless you want to make the `imp` change and want another review of that.

**Pull request merged**